### PR TITLE
added keyboard events for X11

### PR DESCRIPTION
**Pull request by sebastianbaginski**
It is now possible to listen to keyboard events.
@marcusnaslund

### CODE REVIEW

File: source/ui/GuiEvent.ooc [hunk 1]
```diff
@@ -5,16 +5,20 @@ GuiEvent: abstract class {
...
-KeyboardEvent: class extends GuiEvent {
+MouseEvent: class extends InputEvent {
+	init: func (position: IntPoint2D) { super(position) }
+}
+
+KeyboardEvent: class extends InputEvent {
 	_key: Char
 	key ::= this _key
-	init: func (=_key)
+	init: func (position: IntPoint2D, =_key){ super(position) }
```
Comment (marcusnaslund) [inline source/ui/GuiEvent.ooc]: What's `position` in terms of a keyboard event?

**Comment by marcusnaslund** [comment]
Magic complains.

File: source/ui/GuiEvent.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/ui/GuiEvent.ooc]: Same as for mouse event, position of the mouse pointer on the screen while the key was pressed / released.

**Comment by sebastianbaginski** [comment]
updated

File: source/ui/GuiEvent.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/ui/GuiEvent.ooc]: Right, that makes sense, thanks.

**Pull request merged**